src/main/java/edu/greenblitz/tobyDetermined/subsystems/telescopicArm/Elbow.java

```diff
@@ -0,0 +1,110 @@
+package edu.greenblitz.tobyDetermined.subsystems.telescopicArm;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.tobyDetermined.subsystems.GBSubsystem;
+import edu.greenblitz.utils.RoborioUtils;
+import edu.greenblitz.utils.motors.GBSparkMax;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+
+
+public class Elbow extends GBSubsystem {
+
+
+    private static Elbow instance;
+    public ElbowState state = ElbowState.IN_ROBOT;
+    public GBSparkMax motor;
+
+    public static Elbow getInstance() {
+        if (instance == null) {
+            instance = new Elbow();
+        }
+        return instance;
+    }
+
+    private Elbow() {
+        motor = new GBSparkMax(RobotMap.telescopicArm.elbow.MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
+
+        motor.config(new GBSparkMax.SparkMaxConfObject()
+                .withPID(RobotMap.telescopicArm.elbow.PID)
+                .withPositionConversionFactor(RobotMap.telescopicArm.elbow.CONVERSION_FACTOR)
+                .withIdleMode(CANSparkMax.IdleMode.kBrake)
+                .withRampRate(RobotMap.telescopicArm.elbow.MOTOR_RAMP_RATE)
+        );
+        motor.getEncoder().setPosition(0); //resetPosition
+        motor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, RobotMap.telescopicArm.elbow.BACKWARD_ANGLE_LIMIT);
+        motor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, RobotMap.telescopicArm.elbow.FORWARD_ANGLE_LIMIT);
+    }
+
+    public void moveTowardsAngle(double angleInRads) {
+        if (Extender.getHypotheticalState(angleInRads) == Extender.ExtenderState.OUT_OF_BOUNDS) {
+            stop();
+            return;
+        }
+
+        if (Extender.getInstance().getState() == Extender.ExtenderState.OPEN && getHypotheticalState(angleInRads) == ElbowState.IN_ROBOT) {
+            setAngleByPID(RobotMap.telescopicArm.elbow.ENTRANCE_ANGLE);
+        } else {
+            setAngleByPID(angleInRads);
+        }
+    }
+
+    private void setAngleByPID(double goalAngle) {
+        ProfiledPIDController pidController = RobotMap.telescopicArm.elbow.PID_CONTROLLER;
+        pidController.setGoal(goalAngle);
+        double feedForward = getFeedForward(
+                pidController.getSetpoint().velocity, (pidController.getSetpoint().velocity - motor.getEncoder().getVelocity()) / RoborioUtils.getCurrentRoborioCycle());
```

Comment:
this should use the future setpoint (or at the very least previous velocity) as currently it will use accel 0 when velocity is correct